packages/client/src/client.ts

```diff
@@ -161,8 +161,12 @@ export const buildClient = <Plugins extends Record<string, XataPlugin> = {}>(plu
         apiUrl: '',
         // Instead of using workspace and dbBranch, we inject a probably CNAME'd URL
         workspacesApiUrl: (path, params) => {
```

Comment:
@SferaDev this change is very core and needs thorough review. 

When I was calling this function `branch` was `'main'` but I was passing `dbBranchName` of `'dev'` and it still hit main.